lib/source-code/token-store/utils.js

```diff
@@ -49,13 +49,13 @@ exports.getFirstIndex = function getFirstIndex(tokens, indexMap, startLoc) {
     }
     if ((startLoc - 1) in indexMap) {
         const index = indexMap[startLoc - 1];
-        const token = (index >= 0 && index < tokens.length) ? tokens[index] : null;
+        const token = tokens[index];
 
         /*
          * For the map of "comment's location -> token's index", it points the next token of a comment.
          * In that case, +1 is unnecessary.
          */
-        if (token && token.range[0] >= startLoc) {
+        if (!token || token.range[0] >= startLoc) {
```

Comment:
If `token` is undefined, then `index` is already out of bounds, i.e. `index == tokens.length`. In this case, the returned cursor index should point after the last token, at `tokens.length`.
Refs:
- [ForwardTokenCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/forward-token-cursor.js#L34)
- [ForwardTokenCommentCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/forward-token-comment-cursor.js#L35)
- [BackwardTokenCursor](https://github.com/eslint/eslint/blob/27806358b5e1c4d37b63b1c61595e86ff03b5b42/lib/source-code/token-store/backward-token-cursor.js#L35)

Reply:
I may be reading this wrong, but it seems like this change doesn't have any effect. I think it's just refactoring the boolean expression without changing the result. Can you double-check?

Reply:
When `token` is `null` or `undefined`, the new expression evaluates to `true`, whereas the old expression would evaluate to a falsy (`null` or `undefined`) value. Other than this, the result is the same.